### PR TITLE
Add phys damage additional effect procs, add priority additional effects, scripted additional effects, add excalibur (75-119 III) add effect

### LINKED ISSUE #4378: 🐛 Excalibur additional effect not proccing
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my issue will be ignored.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated.

## OS / platform the server is running (if known)

Windows10

## Branch affected by issue 

`base`

## Steps to reproduce

!additem 18276, equip the weapon, fight a few mobs, see the effect never proccing

## Expected behavior

Additional damage effect as per weapon description

I've read open issues about this one, seems like it's been resolved, but for some reason I never see the effect proccing on it? Additional effects do work on the Samurai relic, I haven't tried others. Let me know if I can do some extra testing if needed.

### PR DESCRIPTION
**Pull request by WinterSolstice8**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Bunch of stuff but basically
We needed priority add effect support for Excalibur, and the add effect mods couldn't cut it too well for this edge case so I added scripting support

Fixed add effects not dealing damage or healing (kill me)

Closes #4378
## Steps to test these changes

die in front of the keyboard after testing stuff and realizing damage/healing wasn't being dealt (but now is)

### CODE REVIEW

File: scripts/items/excalibur_119.lua [hunk 1]
```diff
@@ -0,0 +1,24 @@
+-----------------------------------
+-- ID: 20645
+-- Item: Excalibur
+-- Additional effect: Slashing Damage
+-----------------------------------
+---@type TItem
+local itemObject = {}
+
+itemObject.onItemAdditionalEffect = function(attacker, defender, baseAttackDamage, item)
+    if math.random(1, 100) < 7 then -- 7% chance
```
Comment (Xaver-DaRed) [inline scripts/items/excalibur_119.lua]: This is a 6 %

Comment (Xaver-DaRed) [reply scripts/items/excalibur_119.lua]: `if math.random(1, 100) <= 7 then`

**Comment by WinterSolstice8** [comment]
Excalibur proc msg source:

https://youtu.be/I9lN2ixnzck
https://drive.google.com/drive/folders/1f-fhr-m3uhh1msjiwOxOcEaoDqrNvd_w?usp=drive_link

7% rate source (I have seen no data, just numbers)
https://wiki.ffo.jp/html/1479.html

File: scripts/items/excalibur_75.lua [hunk 2]
```diff
@@ -0,0 +1,24 @@
+-----------------------------------
+-- ID: 18276
+-- Item: Excalibur
+-- Additional effect: Slashing Damage
+-----------------------------------
+---@type TItem
+local itemObject = {}
+
+itemObject.onItemAdditionalEffect = function(attacker, defender, baseAttackDamage, item)
+    if math.random(1, 100) <= 7 or true then -- 7% chance
```
Comment (Xaver-DaRed) [inline scripts/items/excalibur_75.lua]: or true?

Comment (WinterSolstice8) [reply scripts/items/excalibur_75.lua]: whoops, debug code made it in, fixed

**Review by Xaver-DaRed — APPROVED** [review]
I think we should in the long run default to the script

But on the long run, not now

File: scripts/items/excalibur_119.lua (hunk 1 above)
Comment (WinterSolstice8) [reply scripts/items/excalibur_119.lua]: fixed
